collage/mainwindow.py

```diff
@@ -121,7 +129,14 @@ def open_text_window(self):
         return window.get_return()
 
     def add_photo(self, where):
-        pass
+        filename = filedialog.askopenfilename(
+            title="Select file",
+            filetypes=(
+                ("jpeg files", "*.jpg"),
+                ("png files", "*.png"),
+            )
+        )
+        self.collage.add_image(filename, where)
```

Comment:
If you press 'Cancel' after you started this dialog, you will get AttributeError, because filename here will be an empty string.

Reply:
You can also add test for this case.